### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by AlexanderZaborski-tomtom**
Implemented the custom endpoint "api/currentuser".

Frontend not completed yet

### CODE REVIEW

File: custom_endpoint/custom_endpoint.routing.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+custom_endpoint.jsonapi:
+  path: '/api/currentuser'
+  defaults:
+    _controller: '\Drupal\custom_endpoint\Controller\CustomEndpointController::renderApi'
+    _title: 'Current user data'
+  methods:  ['GET']
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/custom_endpoint.routing.yml]: unnecessary double space after colon

File: custom_endpoint/custom_endpoint.services.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+services:
+  custom_endpoint.currentUser:
+    class: Drupal\custom_endpoint\CustomEndPointController
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/custom_endpoint.services.yml]: this is not a service, so it is invalid

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Drupal\Core\Session\AccountProxy;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Custom Endpoint Controller class
+ */
+
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: unnecessary empty line

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+class CustomEndpointController extends ControllerBase {
+
+      /**
+     * 
+     * @param AccountInterface $currentUserClass constructor
+     */
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: missing method description
parameters should be typed by the full path

constructor comment block should look like this:

```
  /**
   * Constructs a new object.
   *
   * @param \Drupal\Core\Routing\RouteProviderInterface $route_provider
   *   The route provider service.
   */
```

**Comment by AdamBodera-TomTom** [comment]
General comment: 
fix the comment blocks
every method should have it's return type declared - please fix that :)

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+      //Try to get user data
+      try {
+         $username = $this->currentUser->getAccountName();
+         //$created = $this->currentUser->get('created')->getValue();
+         $created = $this->currentUser->get['created'];
+         $moddate = $this->currentUser->getLastAccessedTime();
+         $lastlogon = $this->currentUser->getLastAccessedTime();
+
+         if ($username == "") {
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: the check if a user is logged in should be done in routing with requirements field
you can also set a specific permission or a role that the user must have to access the endpoint

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      $container->get('current_user'),
+    );
+  }
+
+  /**
+   * @return JsonResponse
+   */
+  public function renderApi(): JsonResponse {
+
+    $currentUserService = \Drupal::service('custom_endpoint.currentUser');
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: now this is a static service invocation - we don't do that
instead dependency inject it the same way as a current_user service

File: custom_endpoint/src/Controller/CustomEndpointController.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  public function __construct(AccountInterface $currentUser) {
+    $this->currentUser = $currentUser;
+  }
+
+  /**
+   * Gets the current user
+   */
+  public static function create(ContainerInterface $container): CustomEndpointController {
+    return new static(
+      $container->get('current_user'),
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/src/Controller/CustomEndpointController.php]: move current_user to the CustomEndpointService, it should be injected there

File: custom_endpoint/custom_endpoint.routing.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+  requirements:
+    _access: 'TRUE'
```
Comment (AdamBodera-TomTom) [inline custom_endpoint/custom_endpoint.routing.yml:8]: this endpoint should not be open - it should be only for the logged in user

File: custom_endpoint/src/Controller/CustomEndpointController.php (hunk 3 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/Controller/CustomEndpointController.php]: Removed empty space

File: custom_endpoint/custom_endpoint.services.yml (hunk 2 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/custom_endpoint.services.yml]: For some reason old code is showing in this comment.
Now class is pointing to \Drupal\custom_endpoint\CustomEndpointService

File: custom_endpoint/custom_endpoint.routing.yml (hunk 1 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/custom_endpoint.routing.yml]: Removed double space

File: custom_endpoint/src/Controller/CustomEndpointController.php (hunk 4 above)
Comment (AlexanderZaborski-tomtom) [reply custom_endpoint/src/Controller/CustomEndpointController.php]: Fixed comment

**Pull request closed without merging**